src/Umbraco.Cms.Api.Delivery/Controllers/Security/MemberController.cs

```diff
@@ -0,0 +1,171 @@
+using Microsoft.Extensions.Options;
+using OpenIddict.Abstractions;
+using OpenIddict.Server.AspNetCore;
+using Umbraco.Cms.Api.Delivery.Routing;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Configuration.Models;
+using Umbraco.Cms.Core.Security;
+using Umbraco.Cms.Web.Common.Security;
+using Umbraco.Extensions;
+using SignInResult = Microsoft.AspNetCore.Mvc.SignInResult;
+using IdentitySignInResult = Microsoft.AspNetCore.Identity.SignInResult;
+
+namespace Umbraco.Cms.Api.Delivery.Controllers.Security;
+
+[ApiVersion("1.0")]
+[ApiController]
+[VersionedDeliveryApiRoute(Common.Security.Paths.MemberApi.EndpointTemplate)]
+[ApiExplorerSettings(GroupName = "Security")]
+public class MemberController : DeliveryApiControllerBase
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IMemberSignInManager _memberSignInManager;
+    private readonly IMemberManager _memberManager;
+    private readonly DeliveryApiSettings _deliveryApiSettings;
+    private readonly ILogger<MemberController> _logger;
+
+    public MemberController(
+        IHttpContextAccessor httpContextAccessor,
+        IMemberSignInManager memberSignInManager,
+        IMemberManager memberManager,
+        IOptions<DeliveryApiSettings> deliveryApiSettings,
+        ILogger<MemberController> logger)
+    {
+        _httpContextAccessor = httpContextAccessor;
+        _memberSignInManager = memberSignInManager;
+        _memberManager = memberManager;
+        _logger = logger;
+        _deliveryApiSettings = deliveryApiSettings.Value;
+    }
+
+    [HttpGet("authorize")]
+    [MapToApiVersion("1.0")]
+    public async Task<IActionResult> Authorize()
+    {
+        // in principle this is not necessary for now, since the member application has been removed, thus making
+        // the member client ID invalid for the authentication code flow. however, if we ever add additional flows
+        // to the API, we should perform this check, so we might as well include it up front.
+        if (_deliveryApiSettings.MemberAuthorizationIsEnabled() is false)
+        {
+            return BadRequest("Member authorization is not allowed.");
+        }
+
+        HttpContext context = _httpContextAccessor.GetRequiredHttpContext();
+        OpenIddictRequest? request = context.GetOpenIddictServerRequest();
+        if (request is null)
+        {
+            return BadRequest("Unable to obtain OpenID data from the current request");
+        }
+
+        // make sure this endpoint ONLY handles member authentication
+        if (request.ClientId is not Constants.OAuthClientIds.Member)
+        {
+            return BadRequest("The specified client ID cannot be used here.");
+        }
+
+        return request.IdentityProvider.IsNullOrWhiteSpace()
+            ? await AuthorizeInternal(request)
+            : await AuthorizeExternal(request);
+    }
+
+    [HttpGet("signout")]
+    [MapToApiVersion("1.0")]
+    public async Task<IActionResult> Signout()
```

Comment:
Same here - no `ProducesResponseType`

Reply:
Nope - same as backoffice. If anything we should hide this controller from Swagger. Noted as a future potential 😄